java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessagingFactory.java

```diff
@@ -428,9 +439,14 @@ public void deregisterForConnectionError(Link link)
 	{
 		this.registeredLinks.remove(link);	
 	}
+	
+	public void scheduleOnReactorThread(final BaseHandler handler) throws IOException
+	{
+		this.getReactorScheduler().invoke(handler);
```

Comment:
>invoke [](start = 29, length = 6)

should the invoke call itself also protected by the reactorLock? I think the inner work inside invoke() needs to be protected since that calls into reactor.schedule or this.ioSignal.sink().write()... both of which needs to be sequenced?